### PR TITLE
Update the GetDimension hlsl builtin for spirv path. In case of sampl…

### PR DESCRIPTION
**Pull request by pmistryNV**
…er, a combined sampled image needs an OpImage to be generated.

Fixes bug #3360

### CODE REVIEW

**Comment by pmistryNV** [comment]
@csyonghe kindly review

File: tests/cross-compile/vk-sampler-getdimension.slang [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  float4 color : SV_Target;
+};
+
+[[vk::binding(1)]] Sampler3D g_Volume;
+
+// Fragment Shader
+[shader("pixel")]
+PSout main(PSin stage, bool isFrontFacing : SV_IsFrontFace)
+{
+  // CHECK: %image{{.*}} = OpImage
```
Comment (csyonghe) [inline tests/cross-compile/vk-sampler-getdimension.slang:23]: This should be CHECk-NOT:

**Review by csyonghe — COMMENTED** [review]
Looks good to me. Need to fix the test failure.

File: tests/cross-compile/vk-sampler-getdimension.slang (hunk 1 above)
Comment (csyonghe) [reply tests/cross-compile/vk-sampler-getdimension.slang:23]: Ah, sorry it should be there, this is a sampler.

File: source/slang/slang-stdlib-textures.cpp [hunk 2]
```diff
@@ -352,16 +369,32 @@ void TextureTypeInfo::writeGetDimensionFunctions()
                     spirv << "%_levelCount : $$uint = OpImageQueryLevels $this;";
                     convertAndStore(UnownedStringSlice("_levelCount"), "numberOfLevels");
                 }
+            };
+            StringBuilder spirvImageVarIn;
```
Comment (csyonghe) [inline source/slang/slang-stdlib-textures.cpp:373]: Line 363 and 369 need changes too.

File: source/slang/slang-stdlib-textures.cpp [hunk 3]
```diff
@@ -271,16 +288,16 @@ void TextureTypeInfo::writeGetDimensionFunctions()
                 glsl << "\";\n";
             }
 
-            StringBuilder spirv;
+            auto generateSpirvAsm = [&](StringBuilder& spirv, StringBuilder& imageVar)
```
Comment (csyonghe) [inline source/slang/slang-stdlib-textures.cpp:291]: This lambda can take in a UnownedStringSlice as parameter instead of a StringBuilder.

**Comment by csyonghe** [comment]
The test failure here seems legit, @pmistryNV can you take another look?

**Comment by pmistryNV** [comment]
> The test failure here seems legit, @pmistryNV can you take another look?

Yes looking into it.

**Comment by pmistryNV** [comment]
@csyonghe kindly review

**Review by csyonghe — APPROVED** [review]

**Pull request merged**